include/crocoddyl/core/actions/lqr.hxx

```diff
@@ -51,6 +51,22 @@ void ActionModelLQRTpl<Scalar>::calc(const boost::shared_ptr<ActionDataAbstract>
       Scalar(0.5) * x.dot(Lxx_ * x) + Scalar(0.5) * u.dot(Luu_ * u) + x.dot(Lxu_ * u) + lx_.dot(x) + lu_.dot(u);
 }
 
+template <typename Scalar>
+void ActionModelLQRTpl<Scalar>::calc(const boost::shared_ptr<ActionDataAbstract>& data,
+                                     const Eigen::Ref<const VectorXs>& x) {
+  if (static_cast<std::size_t>(x.size()) != state_->get_nx()) {
+    throw_pretty("Invalid argument: "
+                 << "x has wrong dimension (it should be " + std::to_string(state_->get_nx()) + ")");
+  }
+
+  if (drift_free_) {
+    data->xnext = x;
+  } else {
+    data->xnext = x + f0_;
```

Comment:
Same comment as above: is it really necessary to update `xnext`?

Reply:
Same answer here.
Done!